full/src/test/java/apoc/custom/CypherProceduresStorageTest.java

```diff
@@ -144,4 +145,12 @@ public void testAllParameterTypesFunction() throws Exception {
         restartDb();
         TestUtil.testCall(db, "return custom.answer(42,3.14,'foo',{a:1},[1],true,date(),datetime(),point({x:1,y:2})) as data", (row) -> assertEquals(9, ((List)row.get("data")).size()));
     }
+
+    @Test
+    public void testIssue1714WithRestartDb() throws Exception {
+        db.executeTransactionally("CREATE (i:Target {value: 2});");
+        db.executeTransactionally("CALL apoc.custom.asFunction('n', 'MATCH (t:Target {value : $val}) RETURN t', 'NODE', [['val', 'INTEGER']])");
+        restartDb();
+        TestUtil.testCall(db, "RETURN custom.n(2) as row", (row) -> assertEquals(2L, ((Node) row.get("row")).getProperty("value")));
```

Comment:
did the test fail before the fixes?

Reply:
Yes, with a `ClassCastException: class java.util.HashMap cannot be cast to class org.neo4j.graphdb.Node`.